### PR TITLE
Fixed #32528 -- Replaced django.utils.topological_sort with graphlib.TopologicalSort().

### PR DESCRIPTION
**Pull request by ngnpope**
ticket-32528.

Supersedes #14102. Blocked on #16467.

### CODE REVIEW

File: tests/migrations/test_autodetector.py [hunk 1]
```diff
@@ -2218,8 +2218,8 @@ def test_circular_fk_dependency(self):
         # Right number/type of migrations?
         self.assertNumberMigrations(changes, "testapp", 1)
         self.assertOperationTypes(changes, "testapp", 0, ["CreateModel", "CreateModel"])
-        self.assertOperationAttributes(changes, "testapp", 0, 0, name="Publisher")
-        self.assertOperationAttributes(changes, "testapp", 0, 1, name="Author")
+        self.assertOperationAttributes(changes, "testapp", 0, 0, name="Author")
+        self.assertOperationAttributes(changes, "testapp", 0, 1, name="Publisher")
```
Comment (ngnpope) [inline tests/migrations/test_autodetector.py:2222]: See https://github.com/django/django/pull/14102#discussion_r590195580 in previous PR.

File: tests/forms_tests/tests/test_media.py [hunk 2]
```diff
@@ -606,7 +606,7 @@ def test_merge(self):
     def test_merge_warning(self):
         msg = "Detected duplicate Media files in an opposite order: [1, 2], [2, 1]"
         with self.assertWarnsMessage(RuntimeWarning, msg):
-            self.assertEqual(Media.merge([1, 2], [2, 1]), [1, 2])
+            self.assertEqual(Media.merge([1, 2], [2, 1], None), [1, 2])
```
Comment (ngnpope) [inline tests/forms_tests/tests/test_media.py:609]: This is just testing the existing behaviour of stripping `None` which was previously untested.

File: django/forms/widgets.py [hunk 3]
```diff
@@ -151,29 +150,23 @@ def merge(*lists):
...
+            return list(ts.static_order())
+        except CycleError:
             warnings.warn(
                 "Detected duplicate Media files in an opposite order: {}".format(
                     ", ".join(repr(list_) for list_ in lists)
                 ),
                 MediaOrderConflictWarning,
             )
-            return list(all_items)
+            return list(dict.fromkeys(chain.from_iterable(filter(None, lists))))
```
Comment (ngnpope) [inline django/forms/widgets.py:169]: This is in a fallback branch, so no point generating it in the default non-cycle path...

**Review by carltongibson — APPROVED** [review]
Hey @ngnpope — thank you for this, lovely. 👍

**Comment by felixxm** [comment]
@ngnpope Thanks :+1:

**Pull request merged**